llama-index-integrations/readers/llama-index-readers-microsoft-sharepoint/llama_index/readers/microsoft_sharepoint/base.py

```diff
@@ -793,7 +807,7 @@ def get_resource_info(self, resource_id: str, **kwargs) -> Dict:
             if (
                 self.attach_permission_metadata
             ):  # changes in access control should trigger a reingestion of the file
-                permissions = self._get_permissions_info(item)
+                permissions = self.get_permission_info(item)
```

Comment:
item is already a path here, but in the method above, you cast it to a path again?

Reply:
Also, you call get_item_from_path above, and then call it again in get_permission_info -- did you mean to do that? (tbh, I'm surprised if this even works lol)

Reply:
ops here is supposed to be self._get_permissions_info